packages/lit-dev-tools-esm/src/build-unversioned-docs.ts

```diff
@@ -0,0 +1,173 @@
+/**
+ * @license
+ * Copyright 2023 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import {fileURLToPath} from 'url';
+import * as pathlib from 'path';
+import * as fs from 'fs';
+import * as fsPromise from 'fs/promises';
+
+const THIS_DIR = pathlib.dirname(fileURLToPath(import.meta.url));
+const REPO_ROOT = pathlib.resolve(THIS_DIR, '..', '..', '..');
+const CONTENT_PKG = pathlib.resolve(REPO_ROOT, 'packages', 'lit-dev-content');
+const SITE_JSON = pathlib.resolve(CONTENT_PKG, 'site', 'site.json');
+
+interface SiteJSON {
+  latestVersion: 'v1' | 'v2';
+}
+
+type EleventyFrontMatterData = string[];
+
+const SITE_LATEST_VERSION = (
+  JSON.parse(fs.readFileSync(SITE_JSON, 'utf8')) as SiteJSON
+).latestVersion;
+const LATEST_VERSION_CONTENT = pathlib.resolve(
+  CONTENT_PKG,
+  'site',
+  'docs',
+  SITE_LATEST_VERSION
+);
+const UNVERSIONED_VERSION_LOCATION = pathlib.resolve(
+  CONTENT_PKG,
+  'site',
+  'docs',
+  'unversioned'
+);
+
+/**
+ * This script builds our unversioned latest documentation for lit.dev. It
+ * locates the current latestVersion of Lit to show from `site.json`, then
+ * copies that documentation from the versioned subdirectory into an unversioned
+ * subdirectory.
+ *
+ * The following transforms are then applied on the files:
+ *   - Permalink frontmatter is added to strip the version from the URL. E.g.,
+ *     /docs/v2/* becomes /docs/*.
+ *  - `api.html` has frontmatter transformed so base api paths become
+ *    unversioned.
+ *  - The latest version `v[N].json` file is renamed `unversioned.json` so it
+ *    matches the directory name which contains it.
+ */
+const buildAndTransformUnverionedDocs = async () => {
+  const walk = async (dirPath: string): Promise<void> => {
+    const entries = await fsPromise.readdir(dirPath, {withFileTypes: true});
+    await Promise.all(
+      entries.map(async (entry) => {
+        const childPath = pathlib.join(dirPath, entry.name);
+        if (entry.isDirectory()) {
+          return walk(childPath);
+        } else {
+          transformFile(childPath);
+        }
+      })
+    );
+  };
+
+  console.log('Starting build-unversioned-docs.js script');
+  await walk(LATEST_VERSION_CONTENT);
+  console.log('Completed build-unversioned-docs.js script');
+};
+
+/**
+ * Transform the given versioned content file and copy it to the unversioned
+ * directory.
+ */
+function transformFile(path: string) {
+  const relativeChildPath = pathlib.relative(LATEST_VERSION_CONTENT, path);
+  const ext = pathlib.extname(relativeChildPath);
+  const relativeChildPathWithoutExt = relativeChildPath.slice(0, -ext.length);
+  const fileBasename = pathlib.basename(relativeChildPath, ext);
+
+  let fileContents = fs.readFileSync(path, {encoding: 'utf8'});
```

Comment:
Is it possible to make this async? That'll make the Promise.all() in `buildAndTransformUnverionedDocs` more effective.

Reply:
Yes! Done!